### ISSUE TITLE
is it still problem or fixed : If you care about predicting the right probability In such a case, you cannot re-balance the datset

### ISSUE THREAD

**Issue by Sandy4321** [dmlc/xgboost#5640]
in practice data always imbalansed , so in most case xgboost can not be used
it needs more explanation why  - "If you care about predicting the right probability In such a case, you cannot re-balance the dataset"
Then we would be able to find solution. then can you share clear code example how bad predicting the right probability for "Balance the positive and negative weights via scale_pos_weight"
at least May you share clear code example of this use of -  Set parameter max_delta_step to a finite number (say 1) to help convergence

For example 
CATBOOST seems to be is free from this conceptual mistake what you have
Since they do not mention this problem or they just not so open as you are?
https://catboost.ai/docs/concepts/python-reference_parameters-list.html

![image](https://user-images.githubusercontent.com/11426119/81227604-a9949f00-8fba-11ea-9d19-7d0c25993b32.png)
 

https://xgboost.readthedocs.io/en/latest/tutorials/param_tuning.html

If you care about predicting the right probability In such a case, you cannot re-balance the dataset

**Comment by hcho3** [comment]
The problem is not specific to XGBoost but other techniques as well, such as logistic regression.

> **Random Oversampling**: All data points from majority and minority training sets are used. Additionally, instances are randomly picked, with replacement, from the minority training set till the desired balance is achieved. Adding the same minority samples might result in overfitting, thereby reducing the generalization ability of the classifier.

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3946903/

In other words, you may underestimate the prediction probability for the training data points whose true label is the minority class.

You could address overfitting by performing stratified cross-validation: https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.StratifiedKFold.html

**Comment by Sandy4321** [comment]
great thanks for soon answer
At least can you share link to algorithm description how xgboost calculates probability of predictions
For example if xgboost calculates that this particular data sample is predicted to have label "YES" instead of label "NO"  with probability/confidence 0.75. 
What is algorithm to calculate this 0.75 ??
